### ISSUE TITLE
Add "use-serving-port" to health-checks

### ISSUE THREAD

**Issue by arnecls** [terraform-providers/terraform-provider-google#3425]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

For easier use of e.g. network endpoint groups, health checks should support the "--use-serving-port" flag, see [[1]](https://cloud.google.com/load-balancing/docs/health-check-concepts#category_and_port_specification).

This is especially interesting for network endpoint groups as they can be created via a kubernetes resource which makes the port and/or name not easily available as a terraform resource.

### New or Affected Resource(s)

* google_compute_health_check
* google_compute_http_health_check
* google_compute_https_health_check

### Potential Terraform Configuration

```tf
# Propose what you think the configuration to take advantage of this feature should look like.
# We may not use it verbatim, but it's helpful in understanding your intent.

resource "google_compute_health_check" "internal-health-check" {
 name = "internal-service-health-check"

 timeout_sec        = 1
 check_interval_sec = 1

 tcp_health_check {
   use-serving-port = "true"
 }
}
```

### References

* [1] https://cloud.google.com/load-balancing/docs/health-check-concepts#category_and_port_specification
* [2] https://github.com/terraform-providers/terraform-provider-google/issues/2263

**Comment by emilymye** [comment]
Note for self/anyone attempting this: Involves adding a boolean field `useServingPort` that sets field `portSpecification` (currently [beta](https://cloud.google.com/compute/docs/reference/rest/beta/healthChecks)) on resource. We want to create this helper field like gcloud did as different values of portSpecification require different fields on resource: 

- `USE_SERVING_PORT` conflicts with port/portName
- ` USE_FIXED_PORT` requires port but providing port alone (i.e. no portSpecification) will do the same thing
- `USE_NAMED_PORT` requires portName but providing portName alone (i.e. no portSpecification) 
 will do the same thing

**Comment by megan07** [comment]
Thanks @arnecls for opening this issue. It's been closed with #4001 and will be released shortly.
